### PR TITLE
Save last used settings in sample app

### PR DESCRIPTION
**Pull request by dryaz**

### CODE REVIEW

**Review by andre-sonect — APPROVED** [review]

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package ch.sonect.sdk.shop.integrationapp
+
+import android.app.Application
+import android.content.Context
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import ch.sonect.sdk.shop.SonectSDK
+
+class MainActivityViewModel(private val app: Application) : AndroidViewModel(app) {
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt]: AndroidViewModel -> ViewModel

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    val state: LiveData<DataState>
+        get() = _state
+
+    private val configRepository: ConfigRepository = ConfigRepository(app.getSharedPreferences("SampleApp", Context.MODE_PRIVATE))
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt]: Move ConfigRepository to constructor

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import android.app.Application
+import android.content.Context
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import ch.sonect.sdk.shop.SonectSDK
+
+class MainActivityViewModel(private val app: Application) : AndroidViewModel(app) {
+    val state: LiveData<DataState>
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt]: add blank line top

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import android.util.Base64
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import ch.sonect.sdk.shop.SonectSDK
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+
+class SdkWrapperActivityViewModel(private val app: Application) : AndroidViewModel(app) {
+    val state: LiveData<DataState>
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt]: add blank line top

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import android.content.Context
+import android.util.Base64
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import ch.sonect.sdk.shop.SonectSDK
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+
+class SdkWrapperActivityViewModel(private val app: Application) : AndroidViewModel(app) {
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt]: AndroidViewModel -> ViewModel

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    val state: LiveData<DataState>
+        get() = _state
+
+    private val configRepository: ConfigRepository = ConfigRepository(app.getSharedPreferences("SampleApp", Context.MODE_PRIVATE))
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt]: Move ConfigRepository to constructor

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+    private val _state: MutableLiveData<DataState> = MutableLiveData()
+
+    private fun getClientId(): String? = (configRepository.get().find { it is Config.ClientId } as? Config.ClientId)?.value
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt]: move functions block after `init` (constructor)

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+    private fun calculateTokenSdk(): String {
+        return Base64.encodeToString(
+            "${getClientId()}:${getClientSecret()}".toByteArray(),
+            Base64.DEFAULT
+        )
+            .replace("\n", "")
+    }
+
+    private fun calculateSignature(): String {
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/SdkWrapperActivityViewModel.kt]: View model isn't the right place for signature calculation

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/ConfigSetBuilder.kt [hunk 9]
```diff
@@ -0,0 +1,67 @@
+package ch.sonect.sdk.shop.integrationapp
+
+import ch.sonect.sdk.shop.SonectSDK
+
+class ConfigSetBuilder {
```
Comment (igor-giziy) [inline sample/src/main/java/ch/sonect/sdk/shop/integrationapp/ConfigSetBuilder.kt]: In that case with kotlin we don't need builder pattern

**Review by igor-giziy — CHANGES_REQUESTED** [review]

File: sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt (hunk 1 above)
Comment (dryaz) [reply sample/src/main/java/ch/sonect/sdk/shop/integrationapp/MainActivityViewModel.kt]: Need context in here for prefs

**Pull request merged**